examples/bigquery-analyze-realtime-reddit-data/scripts/setup.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run in Cloud Shell to set up your project and deploy solution.
+
+echo "===================================================="
+echo " Setting up environment variables ..."
+
+usage() {
+    echo "Usage: [ -i projectId ] [ -r region ] [ -c redditClientId ] [ -u redditUser ]"
+}
+export -f usage
+
+while getopts ":i:r:c:u:" opt; do
+    case $opt in
+        i ) projectId="$OPTARG";;
+        r ) region="$OPTARG";;
+		c ) redditClientId="$OPTARG";;
+		u ) redditUser="$OPTARG";;
+        \?) echo "Invalid option -$OPTARG"
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+
+echo "===================================================="
+echo " Inputs ..."
+echo " Project ID: ${projectId}" 
+echo " Region: ${region}" 
+echo " Reddit Client Id: ${redditClientId}"
+echo " Reddit User: ${redditUser}" 
+
+
+# GCP
+export APP_BUCKET=${projectId}-reddit-stream-app
+
+# reddit
+echo "Please enter reddit client secret:"
+read -s -r REDDIT_CLIENT_SECRET
+echo "Please enter reddit password:"
+read -s -r REDDIT_PASSWORD
+
+echo "===================================================="
+echo " Enabling services ..."
+
+gcloud config set project "$projectId"
+
+gcloud services enable storage-component.googleapis.com 
+gcloud services enable compute.googleapis.com  
+gcloud services enable servicenetworking.googleapis.com 
+gcloud services enable iam.googleapis.com 
+gcloud services enable cloudbilling.googleapis.com
+gcloud services enable bigquery.googleapis.com
+gcloud services enable dataflow.googleapis.com
+gcloud services enable pubsub.googleapis.com
+
+echo "===================================================="
+echo " Setting external IP access ..."
+
+echo "{
+  \"constraint\": \"constraints/compute.vmExternalIpAccess\",
+	\"listPolicy\": {
+	    \"allValues\": \"ALLOW\"
+	  }
+}" > external_ip_policy.json
+
+gcloud resource-manager org-policies set-policy external_ip_policy.json --project="$projectId"
```

Comment:
I'm not a fan of this, maybe just point out in the readme that VMs need a public IP or a Cloud NAT needs to be deployed in the VPC. 
In this script you might just want to verify that the selected VPC has internet connectivity (either via Public IP or Cloud NAT)

Reply:
I'd like to preserve the single-script deployment. Instead of setting externalIP access for ALL VMs, could I pass the solution's single vm?

Reply:
I honestly believe you can just leave it out altogether and if you feel it's necessary add a shout out to the readme. It's going to be more of a gotcha to set the policy this way due to normal users not having the permissions to modify the policy in any real-world GCP org that is going to set the policy to restrict vmExternalIpAccess.

(BTW better also configure it using terraform:  https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project_organization_policy distributing what infrastructure changes are being done across bash and tf is not really great)